src/dbg/debugger.cpp

```diff
@@ -581,7 +583,42 @@ static void cbGenericBreakpoint(BP_TYPE bptype, void* ExceptionAddress = nullptr
     if(*bp.commandText && commandCondition)  //command
     {
         //TODO: commands like run/step etc will fuck up your shit
-        DbgCmdExec(bp.commandText);
+        varset("$breakpointcondition", breakCondition ? 1 : 0, false);
+        varset("$breakpointlogcondition", logCondition, false);
+        _dbg_dbgcmddirectexec(bp.commandText);

```

Comment:
Yes that's a good point. Well I can just merge it and wait for a report.
Probably most users won't be using these advanced features anyway.
On Sat, 25 Jun 2016 at 18:08, Torusrxxx <notifications@github.com> wrote:

> In src/dbg/debugger.cpp
> <https://github.com/x64dbg/x64dbg/pull/803#discussion_r68490296>:
>
> > @@ -581,7 +583,42 @@ static void cbGenericBreakpoint(BP_TYPE bptype, void* ExceptionAddress = nullptr
> >      if(*bp.commandText && commandCondition)  //command
> >      {
> >          //TODO: commands like run/step etc will fuck up your shit
> > -        DbgCmdExec(bp.commandText);
> > +        varset("$breakpointcondition", breakCondition ? 1 : 0, false);
> > +        varset("$breakpointlogcondition", logCondition, false);
> > +        _dbg_dbgcmddirectexec(bp.commandText);
>
> But it is not possible to get the script-controlled condition if async
> execution mode is used.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/x64dbg/x64dbg/pull/803/files/05aadc39a3cf9bc9bc7e55b752342e879af9b42f#r68490296>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACWCmWNb90jXeEv4D4E17QBoobg9ARU_ks5qPVJpgaJpZM4I-X5Q>
> .
>